lib/cylc/task_proxy.py

```diff
@@ -1508,9 +1508,13 @@ def get_state_summary(self):
         """Return a dict containing the state summary of this task proxy."""
         self.summary['state'] = self.state.status
         self.summary['spawned'] = str(self.has_spawned)
-        self.summary['mean_elapsed_time'] = (
-            float(sum(self.tdef.elapsed_times)) /
-            max(len(self.tdef.elapsed_times), 1))
+        count = len(self.tdef.elapsed_times)
+        if count == 0:
+            self.summary['mean_elapsed_time'] = \
+                max(self.summary['execution_time_limit'], 1)
+        else:
+            self.summary['mean_elapsed_time'] = (
+                float(sum(self.tdef.elapsed_times)) / count)
```

Comment:
The 0/1 is the existing behaviour just arguing semantics. I'll see what I can do about not rendering a progress bar in the absence of any mean execution time limit.